h2/src/main/org/h2/value/ValueRow.java

```diff
@@ -163,4 +166,28 @@ public boolean equals(Object other) {
         return true;
     }
 
+    /**
+     * Converts the values in a ValueRow based on the passed column info.
+     * Creates a new instance if any of the contained item must be converted. Otherwise, returns {@code this}.
+     *
+     * @param provider the cast information provider
+     * @param columns the column info list used for the conversation
+     * @return a ValueRow which contains the converted values
+     *
+     * @see Column#convert(CastDataProvider, Value)
+     */
+    public ValueRow convert(CastDataProvider provider, Column[] columns) {
```

Comment:
You need to move this code to some other place. Value classes may not depend on `Column` and other database objects.

Reply:
I created a static method in the `Column` class instead.